tests/compile-fail/environ-gets-deallocated.rs

```diff
@@ -0,0 +1,10 @@
+extern "C" {
+    static environ: *const *const u8;
+}
```

Comment:
Also if we stop providing this on macOS, you'll need another solution. Something like
```rust
#[cfg(target_os="linux")]
fn get_environ() -> *const *const u8 {
  extern "C" {
    static environ: *const *const u8;
  }
  environ
}
#[cfg(target_os="macos")]
fn get_environ() {
  // call _NSGetEnviron
}
```